zebra-chain/src/parameters/network_upgrade.rs

```diff
@@ -350,17 +350,18 @@ impl NetworkUpgrade {
 
     /// Returns the next expected network upgrade after this network upgrade
     pub fn next_upgrade(self) -> Option<Self> {
-        match self {
-            Genesis => Some(BeforeOverwinter),
-            BeforeOverwinter => Some(Overwinter),
-            Overwinter => Some(Sapling),
-            Sapling => Some(Blossom),
-            Blossom => Some(Heartwood),
-            Heartwood => Some(Canopy),
-            Canopy => Some(Nu5),
-            Nu5 => Some(Nu6),
-            Nu6 => None,
-        }
+        Self::iter()
+            .position(|nu| self == nu)
+            .filter(|&p| (p + 1) < NETWORK_UPGRADES_IN_ORDER.len())
+            .map(|p| NETWORK_UPGRADES_IN_ORDER[p + 1])
+    }
+
+    /// Returns the previous network before after this network upgrade
+    pub fn previous_upgrade(self) -> Option<Self> {
+        Self::iter()
+            .position(|nu| self == nu)
+            .filter(|&p| p > 0)
+            .map(|p| NETWORK_UPGRADES_IN_ORDER[p - 1])
```

Comment:
Optional/Nitpick, requires a suggestion below:

```suggestion
        Self::iter().skip_while(|&nu| self != nu).nth(1)
    }

    /// Returns the previous network before after this network upgrade
    pub fn previous_upgrade(self) -> Option<Self> {
        Self::iter().rev().skip_while(|&nu| self != nu).nth(1)
```

Reply:
Amazing. Thanks.